src/coreclr/jit/lower.h

```diff
@@ -217,9 +217,18 @@ class Lowering final : public Phase
         GenTree* oldUseNode = use.Def();
         if ((oldUseNode->gtOper != GT_LCL_VAR) || (tempNum != BAD_VAR_NUM))
         {
-            use.ReplaceWithLclVar(comp, tempNum);
+            GenTree* assign;
+            use.ReplaceWithLclVar(comp, tempNum, &assign);
+
             GenTree* newUseNode = use.Def();
             ContainCheckRange(oldUseNode->gtNext, newUseNode);
+
+            // We need to lower the LclVar and assignment since there may be certain
+            // types or scenarios, such as TYP_SIMD12, that need special handling
+
+            LowerNode(assign);
```

Comment:
I do not think it is a correct approach, this function is used during different phases, for example, during long decomposition, so calling `LowerNode` during it is incorrect. 
`ReplaceWithLclVar` creates such IR:
```
currentNode
storeCurrentNodeToNewNode
NewNodeUse
```
and lowering works node by node, so after we are done with `currentNode` we should go to the next and lower it, why don't we?

See, for example, `LowerSwitch`
https://github.com/dotnet/runtime/blob/b443b8ccce174ef0e87183e21561c4ff13373cfd/src/coreclr/jit/lower.cpp#L504
we call `ReplaceWithLclVar` and it creates these new nodes, but then we return `node->gtNext` that is the created store:
https://github.com/dotnet/runtime/blob/b443b8ccce174ef0e87183e21561c4ff13373cfd/src/coreclr/jit/lower.cpp#L785-L791
and it does their lowering.

Reply:
The one used in long decomposition is the instance method exposed on `LIR::Use` which is distinct from this method that is explicit to lowering and which itself calls into the `LIR::Use` method

Reply:
> and lowering works node by node, so after we are done with currentNode we should go to the next and lower it, why don't we?

We should already have lowered all the previous nodes, we aren't removing the "current" node, and we aren't inserting a node after the "current" node, so the "next" node is already correct. We simply need to ensure new nodes created and inserted before this node are also lowered.

This is how the existing HWIntrinsic lowering has been handled and so the same technique is used here as well.

Reply:
> We should already have lowered all the previous nodes, we aren't removing the "current" node, 

Agree

> and we aren't inserting a node after the "current" node, so the "next" node is already correct.

Don't understand, `ReplaceWithLclVar` creates new nodes and inserts them after the "current" node, what do you mean?

Reply:
> Don't understand, ReplaceWithLclVar creates new nodes and inserts them after the "current" node, what do you mean?

`ReplaceWithLclVar` creates new nodes and inserts them after the node it replaces. In this case, we are replacing an operand of the "current" node, which means it precedes the current node in the list and will already have been lowered.